sound/soc/sof/ipc3-topology.c

```diff
@@ -1979,13 +1979,30 @@ static int sof_ipc3_dai_config(struct snd_sof_dev *sdev, struct snd_sof_widget *
 		return -EINVAL;
 	}
 
-	/*
-	 * DAI_CONFIG IPC during hw_params/hw_free for SSP DAI's is not supported in older
-	 * firmware
-	 */
-	if (v->abi_version < SOF_ABI_VER(3, 18, 0) && config->type == SOF_DAI_INTEL_SSP &&
-	    ((flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) || (flags & SOF_DAI_CONFIG_FLAGS_HW_FREE)))
-		return 0;
+	switch (config->type) {
+	case SOF_DAI_INTEL_SSP:
+		/*
+		 * DAI_CONFIG IPC during hw_params/hw_free for SSP DAI's is not supported in older
+		 * firmware
+		 */
+		if (v->abi_version < SOF_ABI_VER(3, 18, 0) &&
+		    ((flags & SOF_DAI_CONFIG_FLAGS_HW_PARAMS) ||
+		     (flags & SOF_DAI_CONFIG_FLAGS_HW_FREE)))
+			return 0;
+		break;
+	case SOF_DAI_INTEL_HDA:
+		if (data)
+			config->hda.link_dma_ch = data->id;
+		break;
+	case SOF_DAI_INTEL_ALH:
+		if (data) {
+			config->dai_index = data->dai_index;
+			config->alh.stream_id = data->id;
+		}
+		break;
+	default:
+		break;
```

Comment:
can we add debug log for default case for some unexpected types ?

Reply:
its not a error for the other types, they will proceed to go on to do the DAI_CONFIG.